Framework/API/src/AlgorithmProxy.cpp

```diff
@@ -108,21 +109,31 @@ bool AlgorithmProxy::executeAsyncImpl(const Poco::Void &dummy) {
     throw;
   }
   stopped();
-  return m_isExecuted;
+  return (resultState() == ResultState::Success);
```

Comment:
Should this should return `ExecutionState == Finished`?

Reply:
Done as part of the review of the preceding PR, now merged into this one.